addons/mail/static/tests/messaging_menu/messaging_menu.test.js

```diff
@@ -1220,33 +1221,33 @@ test("can open messaging menu even if messaging is not initialized", async () =>
     patchBrowserNotification("default");
     await startServer();
     const def = new Deferred();
-    onRpcBefore("/web/dataset/call_kw/ir.http/lazy_session_info", async () => {
+    const { waitStoreFetch } = listenStoreFetch("init_messaging", async () => {
+        asyncStep("before init_messaging");
         await def;
     });
     await start();
     await click(".o_menu_systray i[aria-label='Messages']");
     await contains(".o-mail-NotificationItem", { text: "Turn on notifications" });
+    await waitForSteps(["before init_messaging"]);
+    def.resolve();
+    await waitStoreFetch("init_messaging");
 });
 
 test("can open messaging menu even if channels are not fetched", async () => {
     patchBrowserNotification("denied");
     const pyEnv = await startServer();
     pyEnv["discuss.channel"].create({ name: "General" });
     const def = new Deferred();
-    onRpcBefore("/mail/action", async (args) => {
-        if (args.fetch_params.includes("channels_as_member")) {
-            await def;
-        }
-    });
-    onRpcBefore("/mail/data", async (args) => {
-        if (args.fetch_params.includes("channels_as_member")) {
-            await def;
-        }
+    const { waitStoreFetch } = listenStoreFetch("channels_as_member", async () => {
+        asyncStep("before channels_as_member");
```

Comment:
Note: this step is necessary to ensure the callback is actually applied, it makes the test much more robust.